cli/internal/terraform/terraform.go

```diff
@@ -33,10 +36,70 @@ import (
 )
 
 const (
+	// LogLevelNone represents a log level that does not produce any output.
+	LogLevelNone LogLevel = iota
+	// LogLevelError enables log output at ERROR level.
+	LogLevelError
+	// LogLevelWarn enables log output at WARN level.
+	LogLevelWarn
+	// LogLevelInfo enables log output at INFO level.
+	LogLevelInfo
+	// LogLevelDebug enables log output at DEBUG level.
+	LogLevelDebug
+	// LogLevelTrace enables log output at TRACE level.
+	LogLevelTrace
+	// LogLevelJSON enables log output at TRACE level in JSON format.
+	LogLevelJSON
 	tfVersion         = ">= 1.2.0"
 	terraformVarsFile = "terraform.tfvars"
 )
 
+// LogLevel is a Terraform log level.
+// As per https://developer.hashicorp.com/terraform/internals/debugging
+type LogLevel int
+
+// ParseLogLevel parses a log level string into a Terraform log level.
+func ParseLogLevel(level string) (LogLevel, error) {
+	switch strings.ToUpper(level) {
+	case "NONE":
+		return LogLevelNone, nil
+	case "ERROR":
+		return LogLevelError, nil
+	case "WARN":
+		return LogLevelWarn, nil
+	case "INFO":
+		return LogLevelInfo, nil
+	case "DEBUG":
+		return LogLevelDebug, nil
+	case "TRACE":
+		return LogLevelTrace, nil
+	case "JSON":
+		return LogLevelJSON, nil
+	default:
+		return LogLevelNone, fmt.Errorf("invalid log level %s", level)
+	}
+}
+
+// String returns the string representation of a Terraform log level.
+func (l LogLevel) String() string {
+	switch l {
+	case LogLevelError:
+		return "ERROR"
+	case LogLevelWarn:
+		return "WARN"
+	case LogLevelInfo:
+		return "INFO"
+	case LogLevelDebug:
+		return "DEBUG"
+	case LogLevelTrace:
+		return "TRACE"
+	case LogLevelJSON:
+		return "JSON"
+	default:
+		return ""
+	}
+}
+
```

Comment:
Maybe this could be done in a new file so the more important code of the client is easier to find.

Reply:
Also you might want to use stringer for this.

Reply:
[This one here](https://pkg.go.dev/golang.org/x/tools/cmd/stringer)?

Reply:
This would mean we'd have to rename the constants to their string value though which I think isn't very intuitive when calling them from other packages (e.g. `terraform.NONE`; What indicates that this is a log level?) 

And we can't change the string values to `"LogLevelNone"` etc. as they come from Terraform

What do you think?